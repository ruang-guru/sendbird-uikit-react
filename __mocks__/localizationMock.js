export const STRING_SET = {
  BUTTON__READ_MORE: 'Baca selengkapnya',
  CHANNEL__MESSAGE_LIST__NOTIFICATION__ON: 'pada',
  CHANNEL__MESSAGE_LIST__NOTIFICATION__NEW_MESSAGE: 'pesan baru',
  LABEL__OPERATOR: 'Guru',
  LABEL__DATE_TODAY: 'Hari ini',
  LABEL__DATE_YESTERDAY: 'Kemarin',
  LABEL__DAY_SUNDAY: 'Minggu',
  LABEL__DAY_MONDAY: 'Senin',
  LABEL__DAY_TUESDAY: 'Selasa',
  LABEL__DAY_WEDNESDAY: 'Rabu',
  LABEL__DAY_THURSDAY: 'Kamis',
  LABEL__DAY_FRIDAY: 'Jumat',
  LABEL__DAY_SATURDAY: 'Sabtu',
};

export default STRING_SET;
