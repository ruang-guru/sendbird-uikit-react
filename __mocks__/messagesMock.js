export const USER_ID_A = 'user_id_a';
export const BASIC_MESSAGE = {
  messageId: 1479690277,
  messageType: 'user',
  channelUrl:
    'sendbird_group_channel_337761017_9f2bf745b4e1d55a20c4b33a8f3a1cb6daecf47d',
  data: '',
  customType: '',
  silent: false,
  createdAt: 1632383106322,
  updatedAt: 0,
  channelType: 'group',
  metaArrays: [],
  reactions: [],
  mentionType: 'users',
  mentionedUsers: [],
  sendingStatus: 'succeeded',
  parentMessageId: 0,
  parentMessageText: null,
  threadInfo: {
    replyCount: 0,
    mostRepliedUsers: [],
    lastRepliedAt: 0,
    updatedAt: 0,
  },
  ogMetaData: null,
  isOperatorMessage: false,
  appleCriticalAlertOptions: null,
  message: 'Welcome to the classroom!',
  sender: {
    nickname: 'Kukuh Aji Sulistyo Bangun Jaya Abadi Mangkubumi',
    plainProfileUrl:
      'https://avatars3.githubusercontent.com/u/46333979?s=460&v=4',
    profileUrl: 'https://avatars3.githubusercontent.com/u/46333979?s=460&v=4',
    userId: 'random-user-id-xxx',
    connectionStatus: 'nonavailable',
    lastSeenAt: 0,
    metaData: {},
    isActive: true,
    friendDiscoveryKey: null,
    friendName: null,
    _preferredLanguages: null,
    requireAuth: false,
    role: 'none',
    isBlockedByMe: false,
  },
  reqId: '1632382917015',
  translations: {},
  requestState: 'succeeded',
  requestedMentionUserIds: [],
  errorCode: 0,
  messageSurvivalSeconds: -1,
  plugins: [],
  poll: null,
};

export const BASIC_MESSAGE_A_1 = {
  ...BASIC_MESSAGE,
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};
export const BASIC_MESSAGE_A_2 = {
  ...BASIC_MESSAGE,
  message:
    'Can you think of other strategies for ensuring a welcoming first day of school? Do you have ideas that you have used successfully over the years? Tell us about them n the comments field below.',
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};
export const BASIC_MESSAGE_A_3 = {
  ...BASIC_MESSAGE,
  message: 'Of course, outside reading log',
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};

export const ASSIGNMENT_MESSAGE_A_1 = {
  ...BASIC_MESSAGE,
  customType: 'assignment',
  data: "{\"serial\":\"ASN-F8ES2S32KFTAR5S6\",\"title\":\"Pilihan Ganda Template with deadline\",\"description\":\"Pilihan Ganda Template\",\"dueAt\":\"2022-01-01T13:20:00Z\",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"cta\":\"ruangguru://ruangkelas?page=assignment_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026assignment_serial=ASN-F8ES2S32KFTAR5S6\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/assignment/detail/ASN-F8ES2S32KFTAR5S6\"}",
};

export const ASSIGNMENT_MESSAGE_A_2 = {
  ...ASSIGNMENT_MESSAGE_A_1,
  data: "{\"serial\":\"ASN-IHB4XI1KY1HWRWPO\",\"title\":\"Pilihan Ganda Template\",\"description\":\"Pilihan Ganda Template\",\"dueAt\":\"\",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"cta\":\"ruangguru://ruangkelas?page=assignment_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026assignment_serial=ASN-IHB4XI1KY1HWRWPO\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/assignment/detail/ASN-IHB4XI1KY1HWRWPO\"}",
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};

export const MATERIAL_MESSAGE_A_1 = {
  ...BASIC_MESSAGE,
  customType: 'material',
  data: "{\"serial\":\"SUBTOPI-2N28ZO17\",\"title\":\"Materi Google\",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"description\":\"\",\"cta\":\"ruangguru://ruangkelas?page=material_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026material_serial=SUBTOPI-2N28ZO17\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/material/detail/SUBTOPI-2N28ZO17\"}"
};

export const MATERIAL_MESSAGE_A_2 = {
  ...MATERIAL_MESSAGE_A_1,
  data: "{\"serial\":\"SUBTOPI-2N28ZO17\",\"title\":\"Understand how users behave on your site, what they need,  Understand how users behave on your site, and how they feel, fast \",\"creatorSerial\":\"JILLYU9G79PUXOHK\",\"description\":\"\",\"cta\":\"ruangguru://ruangkelas?page=material_detail\\u0026workspace_serial=LMS-WS-YPJJS4QTPQU89LR1\\u0026classroom_serial=LMS-CR-ARHSS4XK3RP1R0HQ\\u0026material_serial=SUBTOPI-2N28ZO17\",\"ctaWeb\":\"https://kelas.sirogu.com/workspace/LMS-WS-YPJJS4QTPQU89LR1/classroom/LMS-CR-ARHSS4XK3RP1R0HQ/material/detail/SUBTOPI-2N28ZO17\"}",
  sender: { ...BASIC_MESSAGE.sender, userId: USER_ID_A },
};
export const LONG_MESSAGE = {
  ...BASIC_MESSAGE,
  message:
    'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don\'t look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn\'t anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet.',
};
export const LONG_MESSAGE_A_1 = {
  ...LONG_MESSAGE,
  sender: { ...LONG_MESSAGE.sender, userId: USER_ID_A },
};

export const OPERATOR_MESSAGE = {
  ...BASIC_MESSAGE,
  isOperatorMessage: true,
};
