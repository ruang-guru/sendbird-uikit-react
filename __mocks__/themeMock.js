export default {
  '--sendbird-light-primary-100': '#FFFAE6',

  '--sendbird-light-background-100': '#E6EBF1',

  '--sendbird-light-onlight-01': '#434856',
  '--sendbird-light-secondary-300': '#2EB5C0',
};
