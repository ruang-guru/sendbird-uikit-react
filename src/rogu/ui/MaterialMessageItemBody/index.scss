@import "../../../styles/variables";

.rogu-material-message-item-body {
  position: relative;
  display: inline-block;
  box-sizing: content-box;
  width: 100%;
  min-width: 200px;
  max-width: 400px;

  .rogu-material-message-text {
    white-space: pre-wrap;
    word-break: break-all;
  }

  .rogu-material-message-item-body__text-bubble {
    position: relative;
    padding: 8px 12px;
    box-sizing: border-box;
    border-radius: 16px 16px 0px 0px;
    word-break: break-all;
  }

  .rogu-material-message-item-body__container {
    position: relative;
    overflow: hidden;
    display: flex;
    align-items: center;
    padding: 12px;
    border-radius: 4px;
    cursor: pointer;
    z-index: 10;

    .rogu-material-message-item-body__icon {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    .rogu-material-message-item-body__text-container {
      margin-left: 6px;
    }

    .rogu-material-message-item-body__text-title {
      margin: 0;
    }
  }

  &.outgoing {
    .rogu-material-message-item-body__container {
      @include themed() {
        background-color: t(primary-2);
      }
    }
  }
  &.incoming {
    .rogu-material-message-item-body__container {
      @include themed() {
        background-color: t(bg-1);
      }
    }
  }
}
