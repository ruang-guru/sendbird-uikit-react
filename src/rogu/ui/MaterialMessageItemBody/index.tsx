import React, { ReactElement, useContext } from "react";
import { UserMessage } from "sendbird";
import "./index.scss";

import Label, { LabelTypography, LabelColors } from "../Label";
import { getClassName } from "../../../utils";
import { LocalizationContext } from '../../../lib/LocalizationContext';
import Icon, { IconTypes } from '../Icon';
import { convertCtaLinkToWebLink } from "../../utils";

interface Props {
  className?: string | Array<string>;
  message: UserMessage;
  isByMe?: boolean;
}

export default function MaterialMessageItemBody({
  className,
  message,
  isByMe,
}: Props): ReactElement {
  const { stringSet } = useContext(LocalizationContext);
  const materialData = JSON.parse(message?.data);

  const openMaterial = (): void => {
    if (materialData?.ctaWeb && materialData?.ctaWeb.length > 0){
      window.open(materialData?.ctaWeb);
    } else{
      window.open(convertCtaLinkToWebLink(materialData?.cta, "material"));
    }
  };
  return (
    <div className={getClassName([
      className, 
      "rogu-material-message-item-body", 
      isByMe ? 'outgoing' : 'incoming', 
      message?.reactions?.length > 0 ? 'reactions' : '',
      ])}>
      <div
        className="rogu-material-message-item-body__container"
        onClick={openMaterial}>
        <Icon
            className="rogu-material-message-item-body__icon"
            type={IconTypes.ROGU_MATERIAL}
            width="30"
            height="30"
          />
        <div className="rogu-material-message-item-body__text-container">
          <Label className="rogu-material-message-item-body__text-title" color={LabelColors.ONBACKGROUND_1} type={LabelTypography.SUBTITLE_2}>
            {materialData?.title}
          </Label>
          <div>
            <Label className="rogu-material-message-item-body__text-title" color={LabelColors.ONBACKGROUND_2} type={LabelTypography.BODY_2}>
              {stringSet.MATERIAL}
            </Label>            
          </div>
        </div>
      </div>
    </div>
  );
}
