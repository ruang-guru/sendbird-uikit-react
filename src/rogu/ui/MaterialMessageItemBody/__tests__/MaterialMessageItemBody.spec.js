import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MATERIAL_MESSAGE_A_1 } from '../../../../../__mocks__/messagesMock';

import MaterialMessageItemBody from '../index';

describe('MaterialMessageItemBody', () => {
  it('should do a snapshot test of the MaterialMessageItemBody DOM', function () {
    const text = 'example-text';
    const component = renderer.create(
      <MaterialMessageItemBody message={MATERIAL_MESSAGE_A_1} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
