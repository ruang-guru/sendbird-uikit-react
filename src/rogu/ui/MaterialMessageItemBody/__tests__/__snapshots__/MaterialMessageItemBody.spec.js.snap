// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialMessageItemBody should do a snapshot test of the MaterialMessageItemBody DOM 1`] = `
<div
  className=" rogu-material-message-item-body incoming "
>
  <div
    className="rogu-material-message-item-body__container"
    onClick={[Function]}
  >
    <div
      className="rogu-material-message-item-body__icon sendbird-icon "
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "height": "30",
          "minHeight": "30",
          "minWidth": "30",
          "width": "30",
        }
      }
      tabIndex="0"
    >
      <test-file-stub />
    </div>
    <div
      className="rogu-material-message-item-body__text-container"
    >
      <span
        className="rogu-material-message-item-body__text-title sendbird-label sendbird-label--subtitle-2 sendbird-label--color-onbackground-1"
      >
        Materi Google
      </span>
      <div>
        <span
          className="rogu-material-message-item-body__text-title sendbird-label sendbird-label--body-2 sendbird-label--color-onbackground-2"
        >
          Material
        </span>
      </div>
    </div>
  </div>
</div>
`;
